script/hipify-inplace.sh

```diff
@@ -51,27 +60,47 @@ echo "Logs written to ${HIPIFY_LOG}"
-    sed -i 's@\bDGL_USE_CUDA\b@DGL_USE_ROCM@g' $src
-    sed -i 's@\bCUB_VERSION\b@HIPCUB_VERSION@g' $src
-    sed -i 's@\bCUDART_ZERO_BF16\b@HIPRT_ZERO_BF16@g' $src
-    sed -i 's@\bCUDART_INF_BF16\b@HIPRT_INF_BF16@g' $src
-    sed -i 's@\bthrust::cuda::par@thrust::hip::par@g' $src
-    sed -i 's@\b__nv_fp8_e4m3\b@__hip_fp8_e4m3@g' $src
-    sed -i 's@\b__nv_fp8_e5m2\b@__hip_fp8_e5m2@g' $src
-    sed -i 's@\bCUBLAS_GEMM_DEFAULT_TENSOR_OP\b@HIPBLAS_GEMM_DEFAULT@g' $src
-    sed -i 's@\bcurand4\b@hiprand4@g' $src
-    # hipify uses the old one
-    sed -i 's@\bhip_bfloat16\b@__hip_bfloat16@g' $src
-    sed -i 's@\b__trap();@abort();@' $src
+for src in ${all_srcs[@]}; do
+    sed -i -e 's@#include <hipblas.h>@#include <hipblas/hipblas.h>@' \
+        -e 's@#include <hipsparse.h>@#include <hipsparse/hipsparse.h>@' \
+        -e 's@#include <cuda_fp8.h>@#include <hip/hip_fp8.h>@' \
+        -e 's@#include <cuda_bf16.h>@#include <hip/hip_bf16.h>@' \
+	-e 's@#include <cub/cub.cuh>@#include <hipcub/hipcub.hpp>@' `# TODO: File issue` \
+	-e 's@#include <cuda/functional>@#include <cuda/std/functional>@' `# TODO: File issue` \
+	-e 's@#include "hip/extension/@#include "./cuda/extension/@' `# TODO: File issue` \
+	-e 's@#include "hip/cooperative_minibatching_utils.h"@#include "./cuda/cooperative_minibatching_utils.h"@' `# TODO: File issue` \
+	-e 's@#include "hip/max_uva_threads.h"@#include "./cuda/max_uva_threads.h"@' `# TODO: File issue` \
+	-e 's@\.\./hip/@@' `# TODO: File issue` \
+	-e 's@\bDeviceFor::Bulk\b@Bulk@g' \
+	-e 's@\bcub::@hipcub::@g' `# TODO: File issue` \
+	-e 's@\bcuda::stream_ref@cuco::cuda_stream_ref@' `# TODO: File issue` \
+	-e 's@\bcuda::::min@min@' `# TODO: File issue` \
+	-e 's@\bcuda::proclaim_return_type@proclaim_return_type@' `# TODO: File issue` \
+        -e 's@\bDGL_USE_CUDA\b@DGL_USE_ROCM@g' \
+        -e 's@\b__CUDA_ARCH__\b@__HIP_DEVICE_COMPILE__@g' `# TODO file issue` \
+        -e 's@\bGRAPHBOLT_USE_CUDA\b@GRAPHBOLT_USE_ROCM@g' \
+        -e 's@\bCUB_VERSION\b@HIPCUB_VERSION@g' \
+        -e 's@\bCUDART_ZERO_BF16\b@HIPRT_ZERO_BF16@g' \
+        -e 's@\bCUDART_INF_BF16\b@HIPRT_INF_BF16@g' \
+        -e 's@\bthrust::cuda::par@thrust::hip::par@g' \
+        -e 's@\b__nv_fp8_e4m3\b@__hip_fp8_e4m3@g' \
+        -e 's@\b__nv_fp8_e5m2\b@__hip_fp8_e5m2@g' \
+        -e 's@\bCUBLAS_GEMM_DEFAULT_TENSOR_OP\b@HIPBLAS_GEMM_DEFAULT@g' \
+        -e 's@\bcurand4\b@hiprand4@g' \
+        -e 's@\bhip_bfloat16\b@__hip_bfloat16@g' `# hipify uses the old one` \
+	-e 's@\bnv_bfloat16\b@hip_bfloat16@g' `# TODO: File issue` \
+        -e 's@\b__trap();@abort();@' \
+        -e 's@_hip.h@.h@' -i.sed.prehip $src # Not sure why hipify is changing the import name, though the file name is the original.
 
     # If no changes were made, delete the prehip file.
     if cmp -s "${src}" "${src}.prehip"; then
         echo "Deleting unchanged ${src}.prehip"
         rm "${src}.prehip"
     fi
+    if [ -e "${src}.sed.prehip" ]; then
```

Comment:
So now we've got two versions of the prehip file, one with .prehip and one with .sed.prehip? I think we should end with just a single .prehip file iff any changes were made by any of our hipify tooling (including custom sed commands)

Reply:
I've removed this now. It was a faulty attempt anyway, and I found the correct solution. The issue was that the hipify-tensoradapter and my copy to hipify-graphbolt would remove the `.prehip` files if there were no changes, but sometimes they were still needed if the sed script caused changes. I've now just removed the `os.unlink()` calls and leave it to `hipify-inplace.sh` to remove unneeded `.prehip` files.